modules/broker-amqp/src/main/java/org/wso2/broker/amqp/codec/frames/ConnectionStartOk.java

```diff
@@ -72,46 +74,37 @@ protected void writeMethod(ByteBuf buf) {
     @Override
     public void handle(ChannelHandlerContext ctx, AmqpConnectionHandler connectionHandler) {
         ctx.fireChannelRead((BlockingTask) () -> {
-            Broker broker = connectionHandler.getBroker();
-            SaslServer saslServer;
-            SaslServerBuilder saslServerBuilder = broker.getAuthenticationManager().getSaslMechanisms()
-                    .get(mechanism.toString());
-            try {
-                if (saslServerBuilder != null) {
-                    saslServer = Sasl.createSaslServer(mechanism.toString(), AmqConstant.AMQP_PROTOCOL_IDENTIFIER,
-                            connectionHandler.getConfiguration().getHostName(),
-                            saslServerBuilder.getProperties(), saslServerBuilder.getCallbackHandler());
+            if (authManager.isAuthenticationEnabled()) {
```

Comment:
Shall we use strategy pattern here? we can plug the authentication strategy upfront and pass the relevant auth strategy to this class rather than the auth manager. Maybe that way we can avoid passing authManager to MethodFactory as well. We can pass the authStrategyFactory maybe.